src/streaming/array.writer.cpp

```diff
@@ -400,15 +323,12 @@ zarr::ArrayWriter::flush_()
     }
 
     // compress buffers and write out
-    compress_buffers_();
-    CHECK(flush_impl_());
+    compress_and_flush_();
 
     const auto should_rollover = should_rollover_();
-    if (should_rollover) {
-        rollover_();
-    }
 
     if (should_rollover || is_finalizing_) {
+        rollover_();
```

Comment:
Compared to the old functionality, rollover_() would now be called if should_rollover==false but is_finalizing==true, whereas it wouldn't have before.  I just want to make sure that is intended.

Reply:
Correct. I previously didn't want to call `rollover_()` when finalizing because it would uselessly increment  a counter, but the alternative was to add another check on `is_finalizing_` to call `close_files_()` -- but only when we had a ragged append dimension. This is simpler, and the increment doesn't matter.